From f6ca51ae314c6e72723a109e49193b96d733ce8d Mon Sep 17 00:00:00 2001
From: Stanislaw Gruszka <sgruszka@redhat.com>
Date: Wed, 10 Dec 2014 10:29:18 -0500
Subject: [wireless] iwlwifi: increase DEFAULT_MAX_TX_POWER

Message-id: <1418207389-7899-5-git-send-email-sgruszka@redhat.com>
Patchwork-id: 101577
O-Subject: [RHEL7.1 PATCH 04/35] iwlwifi: increase DEFAULT_MAX_TX_POWER
Bugzilla: 1155538
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: John Greene <jogreene@redhat.com>

From: Eliad Peller <eliad@wizery.com>

BZ#1155538

commit 22d059a5c7c5de61e53c88e30b65e55fbfd91e91 upstream.

The chip is able to transmit up to 22dBm, so set
the constant appropriately.

Signed-off-by: Eliad Peller <eliadx.peller@intel.com>
Signed-off-by: Emmanuel Grumbach <emmanuel.grumbach@intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Brew:
https://brewweb.devel.redhat.com/taskinfo?taskID=8330268

Testing:
Tested on iwl5100, iwl6300, iwl6205, iwl7265 adapters.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/wireless/iwlwifi/iwl-config.h b/drivers/net/wireless/iwlwifi/iwl-config.h
index b704790..6ac1bed 100644
--- a/drivers/net/wireless/iwlwifi/iwl-config.h
+++ b/drivers/net/wireless/iwlwifi/iwl-config.h
@@ -120,6 +120,8 @@ enum iwl_led_mode {
 #define IWL_LONG_WD_TIMEOUT 10000
 #define IWL_MAX_WD_TIMEOUT 120000
 
+#define IWL_DEFAULT_MAX_TX_POWER 22
+
 /* Antenna presence definitions */
 #define ANT_NONE 0x0
 #define ANT_A  BIT(0)
diff --git a/drivers/net/wireless/iwlwifi/iwl-nvm-parse.c b/drivers/net/wireless/iwlwifi/iwl-nvm-parse.c
index 85eee79..0c75fc1 100644
--- a/drivers/net/wireless/iwlwifi/iwl-nvm-parse.c
+++ b/drivers/net/wireless/iwlwifi/iwl-nvm-parse.c
@@ -143,8 +143,6 @@ static const u8 iwl_nvm_channels_family_8000[] = {
 #define LAST_2GHZ_HT_PLUS  9
 #define LAST_5GHZ_HT   161
 
-#define DEFAULT_MAX_TX_POWER 16
-
 /* rate data (static) */
 static struct ieee80211_rate iwl_cfg80211_rates[] = {
  { .bitrate = 1 * 10, .hw_value = 0, .hw_value_short = 0, },
@@ -279,7 +277,7 @@ static int iwl_init_channel_map(struct device *dev, const struct iwl_cfg *cfg,
    * Default value - highest tx power value.  max_power
    * is not used in mvm, and is used for backwards compatibility
    */
-  channel->max_power = DEFAULT_MAX_TX_POWER;
+  channel->max_power = IWL_DEFAULT_MAX_TX_POWER;
   is_5ghz = channel->band == IEEE80211_BAND_5GHZ;
   IWL_DEBUG_EEPROM(dev,
      "Ch. %d [%sGHz] %s%s%s%s%s%s(0x%02x %ddBm): Ad-Hoc %ssupported\n",
-- 
1.7.1